src/plugins/console/public/application/components/refresh_interval.tsx

```diff
@@ -0,0 +1,275 @@
+> {
+  static defaultProps = {
+    isPaused: true,
+    refreshInterval: 1000,
+  };
+
+  state: EuiRefreshIntervalState = fromMilliseconds(this.props.refreshInterval || 0);
+
+  generateId = htmlIdGenerator();
+  legendId = this.generateId();
+  refreshSelectionId = this.generateId();
+
+  onValueChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    const sanitizedValue = parseFloat(event.target.value);
+    this.setState(
+      {
+        value: isNaN(sanitizedValue) ? '' : sanitizedValue,
+      },
+      this.applyRefreshInterval
+    );
+  };
+
+  onUnitsChange: ChangeEventHandler<HTMLSelectElement> = (event) => {
+    this.setState(
+      {
+        units: event.target.value as TimeUnitId,
+      },
+      this.applyRefreshInterval
+    );
+  };
+
+  startRefresh = () => {
+    const { onRefreshChange } = this.props;
+    const { value, units } = this.state;
+
+    if (value !== '' && value > 0 && onRefreshChange !== undefined) {
+      onRefreshChange({
+        refreshInterval: toMilliseconds(units, value),
+        isPaused: false,
+      });
+    }
+  };
+
+  handleKeyDown: KeyboardEventHandler<HTMLElement> = ({ key }) => {
+    if (key === 'Enter') {
+      this.startRefresh();
+    }
+  };
+
+  applyRefreshInterval = () => {
+    const { onRefreshChange, isPaused } = this.props;
+    const { units, value } = this.state;
+    if (value === '') {
+      return;
+    }
+    if (!onRefreshChange) {
+      return;
+    }
+
+    const refreshInterval = toMilliseconds(units, value);
+
+    onRefreshChange({
+      refreshInterval,
+      isPaused: refreshInterval <= 0 ? true : !!isPaused,
+    });
+  };
+
+  toggleRefresh = () => {
+    const { onRefreshChange, isPaused } = this.props;
+    const { units, value } = this.state;
+
+    if (!onRefreshChange || value === '') {
+      return;
+    }
+    onRefreshChange({
+      refreshInterval: toMilliseconds(units, value),
+      isPaused: !isPaused,
+    });
+  };
+
+  render() {
+    const { isPaused } = this.props;
+    const { value, units } = this.state;
+
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    const options = refreshUnitsOptions.find(({ value }) => value === units);
+    const optionText = options ? options.text : '';
+
+    const fullDescription = isPaused ? (
+      <EuiI18n
+        token="euiRefreshInterval.fullDescriptionOff"
+        default="Refresh is off, interval set to {optionValue} {optionText}."
+        values={{
+          optionValue: value,
+          optionText,
+        }}
+      />
+    ) : (
+      <EuiI18n
+        token="euiRefreshInterval.fullDescriptionOn"
+        default="Refresh is on, interval set to {optionValue} {optionText}."
+        values={{
+          optionValue: value,
+          optionText,
+        }}
+      />
+    );
+
+    return (
+      <fieldset>
+        <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false} wrap>
+          <EuiFlexItem grow={false}>
+            <EuiSwitch
```

Comment:
I would suggest you add style: 
```css
  align-items: 'center';
```
to the EuiSwitch as a className. (You can find the example of adding it with `emotion/react` here (https://github.com/elastic/kibana/blob/75f91d7cfabdf7c6133fa288baf7f11832a75555/src/plugins/chart_expressions/expression_tagcloud/public/expression_renderers/tagcloud_renderer.tsx#L62). It would fix the problem of positioning of the label:

Before:

![Screenshot 2021-12-23 at 16 33 02](https://user-images.githubusercontent.com/22456368/147254509-fed43fbc-37cb-4d72-86b2-1d9d3c1858bb.png)

After: 
![Screenshot 2021-12-23 at 16 33 19](https://user-images.githubusercontent.com/22456368/147254534-109052d5-e9f3-4b81-a80e-4b60a07e625b.png)